### PR TITLE
Fix/faucet and rest

### PR DESCRIPTION
**Pull request by poomthiti**
## Describe your changes
1. Retrieve faucet information from faucet url
2. Use api instead of lcd to prevent CORS

### CODE REVIEW

**Review by songwongtp — APPROVED** [review]

File: src/lib/pages/faucet/index.tsx [hunk 1]
```diff
@@ -163,8 +169,8 @@ const Faucet = () => {
       </Heading>
       <Text variant="body2" color="text.dark" pt={4} textAlign="center" mb={8}>
         The faucet provides {faucetAmount} testnet {faucetDenom} per request.
-        Requests are limited to once per hour for each receiving address and IP
-        address.
+        {faucetInfo?.RateLimit &&
+          "Requests are limited to once per hour for each receiving address and IP address."}
```
Comment (bkioshn) [inline src/lib/pages/faucet/index.tsx:173]: Add space in front of Requests " Requests ....."

File: src/lib/app-provider/queries/simulateFee.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ export const useSimulateFeeQuery = ({
     queryFn: async ({ queryKey }) => simulateFn(queryKey[3] as ComposedMsg[]),
     enabled,
     keepPreviousData: true,
-    retry: false,
+    retry: 2,
```
Comment (bkioshn) [inline src/lib/app-provider/queries/simulateFee.ts:69]: Do we have to change for other simulate fee?

Comment (poomthiti) [reply src/lib/app-provider/queries/simulateFee.ts:69]: its not wrapped in useQuery

File: src/lib/pages/faucet/index.tsx [hunk 3]
```diff
@@ -124,7 +126,11 @@ const Faucet = () => {
         setIsLoading(false);
         setResult({
           status: "success",
-          message: `Sent ${faucetAmount} testnet ${faucetDenom} from the faucet. You will need to wait for another hour to request again.`,
+          message: `Sent ${faucetAmount} testnet ${faucetDenom} from the faucet. ${
+            faucetInfo?.RateLimit
+              ? "You will need to wait for another hour to request again."
+              : ""
```
Comment (evilpeach) [inline src/lib/pages/faucet/index.tsx:132]: undefined?

**Review by evilpeach — APPROVED** [review]

**Pull request merged**